libraries/AP_Logger/LogFile.cpp

```diff
@@ -1134,3 +1134,41 @@ void AP_Logger::Write_Winch(bool healthy, bool thread_end, bool moving, bool clu
     };
     WriteBlock(&pkt, sizeof(pkt));
 }
+
+void AP_Logger::Write_PSC(const Vector3f &pos_target, const Vector3f &position, const Vector3f &vel_target, const Vector3f &velocity, const Vector3f &accel_target, const float &accel_x, const float &accel_y)
+{
+    struct log_PSC pkt{
+        LOG_PACKET_HEADER_INIT(LOG_PSC_MSG),
+        time_us         : AP_HAL::micros64(),
+        pos_target_x    : pos_target.x * 0.01f,
+        pos_target_Y    : pos_target.y * 0.01f,
+        position_x      : position.x * 0.01f,
+        position_y      : position.y * 0.01f,
+        vel_target_x    : vel_target.x * 0.01f,
+        vel_target_y    : vel_target.y * 0.01f,
+        velocity_x      : velocity.x * 0.01f,
+        velocity_y      : velocity.y * 0.01f,
+        accel_target_x  : accel_target.x * 0.01f,
+        accel_target_y  : accel_target.y * 0.01f,
+        accel_x         : accel_x * 0.01f,
+        accel_y         : accel_y * 0.01f
+    };
+    WriteBlock(&pkt, sizeof(pkt));
+}
+
+void AP_Logger::Write_PSCP(const Vector3f &vel_error, const Vector2f &vel_xy_p, const Vector2f &vel_xy_i, const Vector2f &vel_xy_d)
+{
+    struct log_PSCP pkt{
+        LOG_PACKET_HEADER_INIT(LOG_PSCP_MSG),
+        time_us     : AP_HAL::micros64(),
+        vel_error_x : vel_error.x * 0.01f,
+        vel_error_y : vel_error.y * 0.01f,
+        vel_xy_p_x  : vel_xy_p.x,
```

Comment:
Could we reorder these fields to have X PID followed by Y PID?  Interleaving them like this is sending me nuts :-)

Reply:
I think it's arguable either way and this would make it inconsistent with the PSC logging.

In general I think I probably agree with PeterB though and I'd say that "PSC" does not match "ATT" for example which has the ordering "DRoll", "Roll", "DPitch", "Pitch".  By the way, in general we log target and actual instead of the error.

Reply:
It is done this way to match PSC. We could reorder both to do all of X then all of Y. I think both orderings will be equally annoying depending on what your trying to do. 

We have target and actual in PSC, we could drop the error and only have the PIDs.

Reply:
Graph x.P x.I and x.D at the same time is going to be more common than graphing x.P y.P....

Reply:
@IamPete1 should we just merge this?

Reply:
I have given in and changed it ;)